pkg/artifacts/artifacts.go

```diff
@@ -17,35 +17,27 @@ type Artifact struct {
 
 // FromResource is a factory method to create an Artifact from an unstructured.Unstructured
 func FromResource(resource unstructured.Unstructured) (*Artifact, error) {
-	var containersNestedKeys []string
+	nestedKeys := getContainerNestedKeys(resource.GetKind())
 
-	switch resource.GetKind() {
-	case k8s.KindPod:
-		containersNestedKeys = []string{"spec", "containers"}
-	case k8s.KindCronJob:
-		containersNestedKeys = []string{"spec", "jobTemplate", "spec", "template", "spec", "containers"}
-	default:
-		containersNestedKeys = []string{"spec", "template", "spec", "containers"}
+	images := make([]string, 0)
+
+	containersImages, err := extractImages(resource, append(nestedKeys, "containers"))
```

Comment:
suggestion : maybe less code  put below instead of `L24-L40` :
```
cTypes := []string{"containers", "ephemeralContainers", "initContainers"}
	for _, t := range cTypes {
		cTypeImages, err := extractImages(resource, append(nestedKeys, t))
		if err != nil {
			return nil, err
		}
		images = append(images, cTypeImages...)
	}
```